### PR TITLE
Additional tests to increase coverage and some minor tweaks to enable testability.

### PR DESCRIPTION
**Pull request by torynet**
Hey John,

We actually made these changes months ago and I never did send you a pull request on them. Essentially we needed better coverage to be able to deploy the ForceDI package and I just felt better with tests. We did have to make some really minor changes, like making some variables test visible and inserting a binding configuration wrapper object so that you could pass in a configuration from the test rather than having to have a configuration in the database.

### CODE REVIEW

**Comment by torynet** [comment]
Now that I have this up here I realize there's a failing test now that worked previously. Apparently something broke in the meantime. Honestly it doesn't make much sense to me, but I also have no idea why we were doing this test...

givenFlowNameWhenReturningThenExceptionThrown is failing on line 54 rather than line 64 where it's supposed to fail, and with a different message...

```
System.TypeException: Invalid type: HelloWorld
Class.Flow.Interview.createFlowByName: line 75, column 1
Class.Flow.Interview.createInterview: line 59, column 1
Class.di_FlowTest.givenFlowNameWhenReturningThenExceptionThrown: line 54, column 1
```

And this is why you do the pull request at the time that you make the edits. Unfortunately that was the last thing on my mind at that time.

I honestly can't see why this code shouldn't work.
```
String flowName = 'HelloWorld';
Flow.Interview flow = Flow.Interview.createInterview(flowName, new Map<String, Object>());
```
But run it anonymous in my scratch org and I get:
```
Line: 2, Column: 28
Variable does not exist: Interview
```

Is this some sort of org shape issue? I'm at a loss.

**Comment by douglascayers** [comment]
Hi @torynet, the error about “Interview” variable doesn’t exist is because your defined a variable named `flow` that shadows the Flow namespace. 

`Flow.Interview flow = ...`

Rename your variable so that it doesn’t shadow any namespace names and it’ll work. 

Same advice as not naming variables same name as any SObject type either for same reason, the Apex compiler can’t tell them apart.

**Comment by torynet** [comment]
Oh, ya. I see. I guess I didn't even notice that. I wonder why it didn't cause them problems when they wrote it... Strange. Thanks Doug.

**Comment by torynet** [comment]
Oh, but actually I now realize that was a different error than the test had been getting anyway. I guess it's treated differently in execute anonymous than in the test.

Having fixed that issue:
```
String flowName = 'HelloWorld';
Flow.Interview testFlow = Flow.Interview.createInterview(flowName, new Map<String, Object>());
```
I just get the original error again:
```
Line: 75, Column: 1
System.TypeException: Invalid type: HelloWorld
```
Which still makes no sense to me. Any idea on that one?

**Comment by ImJohnMDaniel** [comment]
@torynet -- the `di_BindingConfigWrapper` class has access modifiers as global.  What is the reason for that verses public?

**Comment by ImJohnMDaniel** [comment]
@torynet -- I downloaded your changes and saw the same error.  When the `Flow.Interview.createInterview(String, new Map<String, Object>());` and `Flow.Interview.createInterview(String, String, new Map<String, Object>());` methods are called in the unit test, it errors out with the following stacktrace:
```
System.TypeException: Invalid type: FooBar                                          Class.Flow.Interview.createFlowByName: line 75, column 1
Class.Flow.Interview.createInterview: line 59, column 1
Class.di_FlowTest.givenFlowNameWhenReturningThenExceptionThrown: line 43, column 1
```
I tried raising test class API version, tried using the namespace version of the call, tried supplying a item in the map before the call, tried changing the value of the flowName string -- nothing helped.

I am assuming that this is an platform bug.

@afawcett and @douglascayers -- Can you guys check internally to see if this is a bug?  I would be curious if @capeterson is aware of this.

**Comment by afawcett** [comment]
Will take a look.

**Comment by afawcett** [comment]
So according to the docs its working as designed ```If the flow doesn't exist in the current org, a TypeException is thrown.```.  Since System types such as Flow.Interview cannot be mocked with Test.createStub currently, further abstraction is required than is present in di_Flow. Something like [this](https://github.com/afawcett/flowextensions) perhaps? Meanwhile I have updated the PR to comment out this test so as not to hold up getting the others merged.

**Comment by afawcett** [comment]
@ImJohnMDaniel apologies, due to some odd config with my GitHub desktop for this repo my push to PR appears to have merged it. My intention was to add a review comment to share my intent and then have you review before final review. Sorry!

**Comment by ImJohnMDaniel** [comment]
No worries

**Comment by ImJohnMDaniel** [comment]
@torynet -- Can you follow up please on the questions that I was asking above?

**Comment by torynet** [comment]
@ImJohnMDaniel I would love to, but I don't see any questions for me. I'll have to study the comment from Andy in order to come up with a better test. The only questions I see from you were directed to Andy and Doug. I would be happy to answer anything I can though and I'll look at a better test as soon as I have a chance.

**Comment by ImJohnMDaniel** [comment]
Oops.  My Bad.  I had not submitted the review questions yet and you could not see them.

@torynet -- Even though we have already merged this through, can I get you to add comments.  I expect that the responses will lead to a minor tweak.  I just wanted to understand your intent before I make the adjustments.

I appreciate the help.

Cheers!

**Comment by torynet** [comment]
I apologize John. I never even realized these questions were here. I'll try to answer them now.

File: force-di/test/classes/di_BindingTest.cls [hunk 1]
```diff
@@ -56,16 +56,135 @@ private class di_BindingTest {
...
         public Bob(String name) { this.Name = name; }
     }
+    
+    public class Charlie {
+        public String Name {get;private set;}
+        public Charlie() { }
+        public Charlie(String name) { this.Name = name; }
+    }
+    
+    public class Zoro {
```
[the remark this answers is not in the document]
Comment (torynet) [reply force-di/test/classes/di_BindingTest.cls:179]: No. I suppose we were just sloppy and I didn't notice it.

File: force-di/test/classes/di_InjectorTest.cls [hunk 2]
```diff
@@ -26,4 +26,16 @@
 
 @IsTest
 private class di_InjectorTest {
-}
+    
+    //TODO
```
[the remark this answers is not in the document]
Comment (torynet) [reply force-di/test/classes/di_InjectorTest.cls:30]: Well, the test kinda sucks. It gets you coverage but it doesn't really test anything. So, yes I think there is more to do here, but just in that a better test that asserts something would be better. Clearly we were lazy and left it as a coverage test.

**Pull request merged**